motionplan/constraint.go

```diff
@@ -130,11 +130,14 @@ func (c *ConstraintHandler) CheckStateConstraintsAcrossSegment(ci *ik.Segment, r
 
 	for i := 0; i <= steps; i++ {
 		interp := float64(i) / float64(steps)
-		interpConfig := referenceframe.InterpolateInputs(ci.StartConfiguration, ci.EndConfiguration, interp)
-		interpC := &ik.State{Frame: ci.Frame, Configuration: interpConfig}
-		err = resolveStatesToPositions(interpC)
-		if err != nil {
-			return false, nil
+		interpC := &ik.State{Frame: ci.Frame}
```

Comment:
This function changed because we need to be able to CheckSegmentAndStateValidity either through start and end configuration or through start and end position.

We introduce Checking SegmentAndStateValidity exclusively through start and end position since that can tell us how much a base has deviated from its path. If a base has deviated from a path, this information is not represented in start and end configuration.